src/umpire/interface/c_fortran/typesUmpire.h

```diff
@@ -26,17 +28,35 @@ struct s_umpire_allocator {
 };
 typedef struct s_umpire_allocator umpire_allocator;
 
-struct s_umpire_dynamicpool {
+struct s_umpire_resourcemanager {
     void *addr;     /* address of C++ memory */
     int idtor;      /* index of destructor */
 };
-typedef struct s_umpire_dynamicpool umpire_dynamicpool;
+typedef struct s_umpire_resourcemanager umpire_resourcemanager;
 
-struct s_umpire_resourcemanager {
+struct s_umpire_strategy_allocationadvisor {
     void *addr;     /* address of C++ memory */
     int idtor;      /* index of destructor */
 };
-typedef struct s_umpire_resourcemanager umpire_resourcemanager;
+typedef struct s_umpire_strategy_allocationadvisor umpire_strategy_allocationadvisor;
+
+struct s_umpire_strategy_dynamicpool {
```

Comment:
While here, should we also create an interface to `DynamicPoolList`?  I think we should.